### PR TITLE
Implemented strategy Cave (k49r from Axelrod's Second)

### PR DESCRIPTION
**Pull request by gaffney2010**
Fingerprints below:

basic_fortran_cave
basic_python_cave
prob_fortran_cave
prob_python_cave

### CODE REVIEW

**Comment by gaffney2010** [comment]
Noticed a slight difference in the fingerprints.  Found that the Fortran was calculating percentage wrong.  Put the same logic in my code.  Fingerprints match better now.
basic_fortran_cave
basic_python_cave
prob_fortran_cave
prob_python_cave

**Comment by gaffney2010** [comment]
Fixed one of my tests for the calculation issue I fixed in last commit.

File: axelrod/tests/strategies/test_axelrod_second.py [hunk 1]
```diff
@@ -490,3 +490,55 @@ def test_strategy(self):
...
+        self.versus_test(axelrod.Cooperator(), expected_actions=actions)
+
+        # It will take until turn 18 to respond decide to repond D->D
+        random.seed(1)
+        actions = [(C, D)]
+        actions += [(C, D), (D, D), (D, D), (C, D), (C, D), (C, D), (D, D),
+                    (D, D), (C, D), (C, D), (D, D), (C, D), (D, D), (C, D),
+                    (C, D), (D, D), (C, D)] # Randomly choose
+        actions += [(D, D)] * 30 # Defect
+        self.versus_test(axelrod.Defector(), expected_actions=actions)
```
Comment (drvinceknight) [inline axelrod/tests/strategies/test_axelrod_second.py]: Can we put the seed argument in to `self.versus_test` please:

```
self.versus_test(axelrod.Defector(), expected_actions=actions, seed=1)
```

Same below.

**Comment by drvinceknight** [comment]
I have gone through the logic and agree this strategy is correctly implemented (as well as the great match of the fps).

**Comment by drvinceknight** [comment]
This is looking good to me. Thanks again @gaffney2010!

**Pull request merged**